make_report.py

```diff
@@ -29,6 +30,30 @@ def build(self) -> List[Section]:
         return self._sections
 
 
+def get_image_caption(filename: str)  -> str:
+    name_without_ext, ext = os.path.splitext(filename)
+
+    parts = name_without_ext.split("_")
+    regex = "^(day)*[0-9]+\.*[0-9]*$"
```

Comment:
Should be raw string, as in
```python
regex = r"^(day)*[0-9]+\.*[0-9]*$""
```
Otherwise python thinks `"\."` is an escape sequence so wants to interpret it as a single character in the same was as ``"\n" would be a single character. I believe in this case it still works because `"\."` is an invalid escape sequence, but normally regexes are specified as raw strings to avoid confusion.

This would also match things like `daydayday2`, `3.`, `3........`, `3..............4324`. If that's intentional then fine, otherwise this might make more sense? (only allow one day, and treat the full stop plus following numbers as one) - may also be helpful to be more strict if the next step is to parse it as a way of ordering the images/assigning them to a report section.
```
r"^(day)?[0-9]+(\.[0-9]+)?$"
```